From: Joel Selvaraj <joelselvaraj.oss@gmail.com>
Date: Thu, 2 May 2024 22:49:47 +0000
Subject: HACK: arm64: dts: qcom: xiaomi-beryllium: enable serial debug

Marked as hack because it adds a 2s delay during the boot up process which helps workaround the random display failure during some reboots.
Issue is discussed here: https://gitlab.freedesktop.org/drm/msm/-/issues/46
and here: https://gitlab.com/postmarketOS/pmaports/-/issues/2518

But, let's keep this even if the bug is fixed, because I got serial debugging work reliably in my Poco F1 atlast!!


(cherry picked from commit 5219473b31eae61ffb3370c1c42d5b49ec9e6a19)

Co-authored-by: Joel Selvaraj <joelselvaraj.oss@gmail.com>
---
 .../dts/qcom/sdm845-xiaomi-beryllium-common.dtsi     | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi b/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi
index 8d266ef..e2864c0 100644
--- a/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi
@@ -38,9 +38,14 @@ / {
 	qcom,msm-id = <321 0x20001>;
 
 	aliases {
+		serial0 = &uart9;
 		serial1 = &uart6;
 	};
 
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
 	gpio-keys {
 		compatible = "gpio-keys";
 		autorepeat;
@@ -466,6 +471,16 @@ &qupv3_id_1 {
 	status = "okay";
 };
 
+&qup_uart9_rx {
+	drive-strength = <2>;
+	bias-pull-up;
+};
+
+&qup_uart9_tx {
+	drive-strength = <2>;
+	bias-disable;
+};
+
 &sdhc_2 {
 	status = "okay";
 
@@ -664,6 +679,11 @@ bluetooth {
 	};
 };
 
+&uart9 {
+	label = "LS-UART1";
+	status = "okay";
+};
+
 &ufs_mem_hc {
 	status = "okay";
 
